### ISSUE TITLE
Unable to find the browser

### ISSUE THREAD

**Issue by darlanmendonca** [DevExpress/testcafe#1330]
### Are you requesting a feature or reporting a bug?
a bug

### What is the current behavior?
run my test in browserstack, but the CLI dont found the name of browser.
I'm trying use the browser `browserstack:chrome@56.0:OS X Sierra`
The cli return the following error

```sh
Error Unable to find the browser. "browserstack:chrome@56.0:OS" is not a browser alias or path to an executable file
```

note, only for the first space, missing the " X Sierra", problably a error on parse values from process.argv

**Comment by AlexanderMoskovkin** [comment]
Hi,

Do you wrap the browser alias with quotes? 

It should be
```sh
testcafe "browserstack:chrome@56.0:OS X Sierra" test.js
```
to be properly parsed by CLI

**Comment by darlanmendonca** [comment]
@AlexanderMoskovkin oh yes, I wrap with quotes

**Comment by inikulin** [comment]
@darlanmendonca You mean that you wrapped browser name with quotes as @AlexanderMoskovkin suggested and problems still persists? Because judging from your error message it's not wrapped.

**Comment by darlanmendonca** [comment]
This, I was already using quotes, and also had already tried without. And the error occurs in both cases

**Comment by AlexanderMoskovkin** [comment]
Please try to run the `testcafe -b browserstack`.
If it will be an error, check that `testcafe` and `testcafe-browser-provider-saucelabs` are installed (both globally or both locally) as I've described [here](https://github.com/ryx/testcafe-browser-provider-nightmare/issues/6#issuecomment-283675843)

**Comment by darlanmendonca** [comment]
@AlexanderMoskovkin first thanks, we need try all, :D

but I already do it this, I get the name from this command `testcafe -b browserstack`

testcafe and testcafe-browser-provider-browserstack exists too, in my node_modules (locally only)

I really guess that is a bug, because on parse process.argv, they split the script (a string), by space, and we dont have flags like `--browser`, or `--files`, just a string separated by space

**Comment by AlexanderMoskovkin** [comment]
@darlanmendonca thanks for the info.

Let's check one more assumption. Could you please install the latest testcafe version with the `alpha` tag and check the issue with it?

```sh
npm install testcafe@alpha
testcafe "browserstack:chrome@56.0:OS X Sierra" test.js
```

Please let me know what the error do you see after that, the same or some other?

We've fixed #1282 issue (it was related to browserstack provider), it possible it will help in your case.

**Comment by inikulin** [comment]
@darlanmendonca Which command prompt/terminal you are using?

**Comment by darlanmendonca** [comment]
@AlexanderMoskovkin I tried now the alpha version, but problem persists :(

same error

**Comment by darlanmendonca** [comment]
@inikulin my script is that

```sh
./node_modules/.bin/testcafe 'browserstack:chrome@56.0:OS X Sierra' \
./sources/scripts/*.spec.js \
--app './node_modules/.bin/http-server . -s'
```

browserstack allow to use last chrome, just using "browserstack:chrome", but to use an iphone for example, we need to fix this

**Comment by AndreyBelym** [comment]
@darlanmendonca You can try to prepend the spaces in browser name with backslashes, like
```
testcafe "browserstack:chrome@56:OS\ X\ Sierra" test.js
```

**Comment by darlanmendonca** [comment]
@AndreyBelym I tried now, same error, oth in the latest version and in the alpha of testcafe :(

**Comment by inikulin** [comment]
@darlanmendonca Can you please post here result of `testcafe -b browserstack` and error that you currently see after running command from https://github.com/DevExpress/testcafe/issues/1330#issuecomment-287751267?

**Comment by darlanmendonca** [comment]
and error about running is 

![screen shot 2017-03-20 at 10 23 36](https://cloud.githubusercontent.com/assets/876227/24101469/367ab114-0d57-11e7-99a7-78c5e290642f.png)

**Comment by darlanmendonca** [comment]
hey guys maybe be my approach passing a variable

my full script is that 

```sh
#!/bin/bash

set -e

: ${browser:=nightmare}

./node_modules/.bin/testcafe $browser \
./sources/scripts/*.spec.js \
--app './node_modules/.bin/http-server . -s'
```

when I give a echo in variable, I receive the right output, but the process dont recognize the full string, like errors that I mention above

now, if I pass directly, with escapes, like

```sh
./node_modules/.bin/testcafe browserstack:chrome@56:OS\ X\ Sierra \
./sources/scripts/*.spec.js \
--app './node_modules/.bin/http-server . -s'
```

works fine :D

**Comment by darlanmendonca** [comment]
the problem now is can pass a environment variable to script

**Comment by AndreyBelym** [comment]
In this case you can try to surround `$browser` with double quotes (`"`) in your script.

**Comment by darlanmendonca** [comment]
yeahh, that works @AndreyBelym 

my script now

```sh
: ${browser:=nightmare}

./node_modules/.bin/testcafe "$browser" \
./sources/scripts/*.spec.js \
--app './node_modules/.bin/http-server . -s'
```

thanks guys, in the end, is just an user (me) error :D

**Comment by AndreyBelym** [comment]
Thank you for using TestCafe 😉 👍

**Comment by darlanmendonca** [comment]
thanks for all, testcafe and help :D

**Comment by amirse80** [comment]
Hi, I am trying to run a test via test runner and I get the same error.
node tRunner.js chrome "C:\\Program Files (x86)\\Google\\Chrome\\Application\\chrome.exe"

I have tested many different combinations like:
1) C:\\...
2) C:\..
3) with surrounding "C:\\" and also without " "
4) I have tested with only "node tRunner.js chrome"

Regardless what I do I always get Error: Unable to find the browser. "path..." is not a browser alias or path to an executable file.

Actually I am trying to run this command from TFS and the tests are run remotly on a SELENIUM_SERVER. So the tests are written with testcafe but I would like to run them on a selenium grid from TFS.

Any suggestion on how to write the "node tRunner.js chrome" command to be able to start/find the browser?

**Comment by AndreyBelym** [comment]
@amirse80, could you please share the code of the `tRunner.js` file?

**Comment by amirse80** [comment]
@AndreyBelym, thanks for your reply. We have found the issue actually. The problem was not in testcafe for some reason not being able to open the browser. Insted it was actually a network issue. We needed to open up a port in our internal network, 1337, and then it started to work. Now I have another issue that is about the browser not being closed after a test (opened with TestCafe DISCONNECTED message). I will check if there are any posts about that. I see actually that it comes to my RUNNER clas where both testcafe.close() and process.exit(0) are stated but it never closes the browser
